### PR TITLE
CRUD operations for snowflake databases

### PR DESCRIPTION
**Pull request by ryanking**
This will CRUD snowflake databases.

Auth is currently username + password only. Will look into okta later.

### CODE REVIEW

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -18,14 +26,68 @@ func Provider() *schema.Provider {
...
+
+func DSN(s *schema.ResourceData) (string, error) {
+	account := s.Get("account").(string)
+	username := s.Get("username").(string)
+	password := s.Get("password").(string)
+	region := s.Get("region").(string)
+	role := s.Get("role").(string)
+
+	// us-west-2 is their default region, but if you actually specify that it won't trigger their default code
+	//  https://github.com/snowflakedb/gosnowflake/blob/52137ce8c32eaf93b0bd22fc5c7297beff339812/dsn.go#L61
```
Comment (edulop91) [inline pkg/provider/provider.go]: :cry:

File: pkg/provider/provider.go [hunk 2]
```diff
@@ -18,14 +26,68 @@ func Provider() *schema.Provider {
...
+		},
 		DataSourcesMap: map[string]*schema.Resource{},
+		ConfigureFunc:  ConfigureProvider,
+	}
+}
+
+func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
+	dsn, err := DSN(s)
+
+	log.Printf("[DEBUG] connecting to %s", dsn)
```
Comment (edulop91) [inline pkg/provider/provider.go]: why not logrus.Debug?

File: pkg/provider/provider_test.go [hunk 3]
```diff
@@ -4,12 +4,64 @@ import (
 	"testing"
 
 	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider"
+	"github.com/hashicorp/terraform/helper/schema"
+	_ "github.com/snowflakedb/gosnowflake"
```
Comment (edulop91) [inline pkg/provider/provider_test.go]: just curious, what is being init'd here?

File: pkg/resources/database.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+func (d *database) Create(data *schema.ResourceData, meta interface{}) error {
+	name := data.Get("name").(string)
+	comment := data.Get("comment").(string)
+	db := meta.(*sql.DB)
+
+	// TODO escape name
+	// TODO escape comment
+	// TODO name appears to get normalized to uppercase, should we do that? or maybe just consider it
+	// 	case-insensitive?
+	stmt := fmt.Sprintf("CREATE DATABASE %s COMMENT='%s'", name, comment)
```
Comment (edulop91) [inline pkg/resources/database.go]: would this allow people to do terrible things?

File: pkg/resources/database.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	log.Printf("[DEBUG] stmt %s", stmt)
+	_, err := db.Exec(stmt)
+	if err != nil {
+		return errors.Wrapf(err, "error dropping database %s", name)
+	}
+
+	return nil
+}
+
+func (d *database) Update(data *schema.ResourceData, meta interface{}) error {
```
Comment (edulop91) [inline pkg/resources/database.go]: neat

File: pkg/provider/provider.go (hunk 2 above)
Comment (ryanking) [reply pkg/provider/provider.go]: That would work if I can get the format right (has to start with "[DEBUG]" for terraform to pass it through correctly.

File: pkg/provider/provider.go (hunk 1 above)
Comment (ryanking) [reply pkg/provider/provider.go]: yeah that was an hour of my life I want back

File: pkg/provider/provider_test.go (hunk 3 above)
Comment (ryanking) [reply pkg/provider/provider_test.go]: The sql driver for snowflake. https://github.com/snowflakedb/gosnowflake/blob/master/driver.go#L141

File: pkg/resources/database.go (hunk 4 above)
Comment (ryanking) [reply pkg/resources/database.go]: Without escaping, yes.

File: pkg/provider/provider.go (hunk 2 above)
Comment (ryanking) [reply pkg/provider/provider.go]: I think I will leave it as-is for now.

File: pkg/resources/database.go (hunk 4 above)
Comment (ryanking) [reply pkg/resources/database.go]: I will follow up with a PR just on escaping. Need to research more.

File: pkg/provider/provider.go (hunk 2 above)
Comment (edulop91) [reply pkg/provider/provider.go]: ah, interesting

**Pull request merged**